File: 프로그래머스/src/프로그래머스_02242022/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    vector<int> temp;
    for(int i = 0 ; i <commands.size() ; i++ )
    {
        int start = commands[i][0];//i
        int end = commands[i][1];//j
        int index = commands[i][2];//K
        
        for(int j = start-1 ; j < end ; j ++ )
        {
            temp.push_back(array[j]); //범위에드는 원소만
        }
        //정렬
       sort(temp.begin(), temp.end());
       answer.push_back(temp[index-1]);
       temp.clear();//초기화
    }
    return answer;
}

File: 2026/Level1/k번째수.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42748?language=python3
def find_key(arr, loc):
    # print("subarray : ", arr)
    arr.sort()
    # print ("sorted array : ", arr)
    key = 0
    for x in range(len(arr)):
        if x == loc-1 :
            print (arr[x])
            return arr[x]
    return key

def solution(array, commands):
    answer = []
    for x in commands :
        # print(x)
        key = find_key(array[x[0]-1:x[1]], x[2])
        answer.append(key)
        
    return answer